internal/permissions/groups/middleware.go

```diff
@@ -0,0 +1,90 @@
+package userpermissions
```

Comment:
Why is the package `userpermissions`? I would think `internal/usergroups/middleware.go` (`usergroups` package) would make more sense.

`userpermissions` handles the access-control from a known role, not the fetching of the role itself.